src/uct/ib/base/ib_md.c

```diff
@@ -328,64 +340,85 @@ void *uct_ib_md_thread(void *arg) {
 ucs_status_t
 uct_ib_md_handle_mr_list_multithreaded(uct_ib_md_t *md, void *address,
                                        size_t length, uint64_t access,
-                                       int mr_num, struct ibv_mr **mrs)
+                                       size_t chunk, struct ibv_mr **mrs)
 {
-    int per_thread, threads, i, mr_idx = 0, cpu = 0;
-    ucs_status_t status = UCS_OK;
+    int thread_num_mrs, thread_num, thread_idx, mr_idx = 0, cpu_id = 0;
+    int mr_num = ucs_div_round_up(length, chunk);
+    ucs_status_t status;
     void *thread_status;
     cpu_set_t parent_set, thread_set;
-    uct_ib_md_thread_ctx_t *ctx;
+    uct_ib_md_mem_reg_thread_t *ctxs, *cur_ctx;
     pthread_attr_t attr;
     int ret;
 
-    pthread_getaffinity_np(pthread_self(), sizeof(cpu_set_t), &parent_set);
-    threads = CPU_COUNT(&parent_set);
+    ucs_assert(pthread_getaffinity_np(pthread_self(), sizeof(cpu_set_t),
```

Comment:
this is no good way to check return value:
1. its compile out in release mode
2. need to print error and exit the function, not abort the process